pkg/models/terminal/terminal.go

```diff
@@ -200,8 +222,12 @@ func (t *terminaler) HandleSession(shell, namespace, podName, containerName stri
 	var err error
 	validShells := []string{"sh", "bash"}
 
+	stopChan := make(chan struct{})
```

Comment:
Use stopCha created by apiserver https://github.com/kubesphere/kubesphere/blob/master/cmd/ks-apiserver/app/server.go#L57

Reply:
I think the lifecycle of the heartbeat channel should not be controlled by apiserver. If the user closes the terminal page, the server shouldn't send heartbeats anymore, so we close the channel to tell the goroutine to stop.